File: trunk/dotnet/picocontainer/src/Test/Defaults/CachingComponentAdapterFactoryTestCase.cs
using System;
using PicoContainer;
using PicoContainer.Defaults;
using PicoContainer.Tests.Tck;
using PicoContainer.Tests.TestModel;
using NUnit.Framework;

namespace Test.Defaults
{
	/// <summary>
	/// Summary description for CachingComponentAdapterFactoryTestCase.
	/// </summary>
	[TestFixture]
	public class CachingComponentAdapterFactoryTestCase : AbstractComponentAdapterFactoryTestCase
	{
		[SetUp]
		protected void setUp()
		{
			picoContainer = new DefaultPicoContainer(CreateComponentAdapterFactory());
		}

		protected override IComponentAdapterFactory CreateComponentAdapterFactory()
		{
			return new CachingComponentAdapterFactory(new ConstructorInjectionComponentAdapterFactory());
		}

		[Test]
		public void testContainerReturnsSameInstaceEachCall()
		{
			picoContainer.RegisterComponentImplementation(typeof (Touchable), typeof (SimpleTouchable));
			Touchable t1 = (Touchable) picoContainer.GetComponentInstance(typeof (Touchable));
			Touchable t2 = (Touchable) picoContainer.GetComponentInstance(typeof (Touchable));
			Assert.AreSame(t1, t2);
		}


	}
}

File: java/picocontainer/src/test/org/picocontainer/defaults/CachingComponentAdapterFactoryTestCase.java
package org.picocontainer.defaults;

import org.picocontainer.tck.AbstractComponentAdapterFactoryTestCase;
import org.picocontainer.testmodel.SimpleTouchable;
import org.picocontainer.testmodel.Touchable;

/**
 * @author <a href="Rafal.Krzewski">rafal@caltha.pl</a>
 * @version $Revision$
 */
public class CachingComponentAdapterFactoryTestCase extends AbstractComponentAdapterFactoryTestCase {
    protected void setUp() throws Exception {
        picoContainer = new DefaultPicoContainer(createComponentAdapterFactory());
    }

    protected ComponentAdapterFactory createComponentAdapterFactory() {
        return new CachingComponentAdapterFactory(new ConstructorComponentAdapterFactory());
    }

    public void testContainerReturnsSameInstaceEachCall() {
        picoContainer.registerComponentImplementation(Touchable.class, SimpleTouchable.class);
        Touchable t1 = (Touchable) picoContainer.getComponentInstance(Touchable.class);
        Touchable t2 = (Touchable) picoContainer.getComponentInstance(Touchable.class);
        assertSame(t1, t2);
    }
}